### ISSUE TITLE
fix(rules): wai-aria false positives on role overwriting for input elements

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3588]
## Describe the bug

The `wai-aria` rule incorrectly reports role overwriting as forbidden on `<input>` elements in cases where the ARIA in HTML specification permits it. For example, `<input type="text" role="combobox">` is valid per the spec, but markuplint reports:

```
Cannot overwrite the "combobox" role to the "input" element according to ARIA in HTML specification
```

The rule also incorrectly reports `aria-checked` as disallowed on `<button>` with certain roles, and has issues with conditional role overwriting rules based on `input[type]`.

## Code Example

```html
<input type="text" role="combobox">
<input type="checkbox" role="switch">
<button role="menuitemcheckbox" aria-checked="true">Toggle</button>
```

## The current behavior

Multiple false positive violations for valid ARIA role/state combinations.

## The expected behavior

No violations when the role overwrite is permitted by the ARIA in HTML specification.

## Impact

Causes ~200 false positives in the nu-html-checker compatibility benchmark (largest single source of false positives).

## Affected package

`@markuplint/rules` — `wai-aria` rule

**Comment by YusukeHirao** [comment]
May benefit from #3645 (wai-aria rule split into sub-rules). Role overwrite checks could have individual severity configuration.

**Comment by YusukeHirao** [comment]
### Benchmark update (58.3%)

wai-aria false positives remain at 335 violations across ~200 files. These are the largest single source of false positives. Still blocked by #3645 (wai-aria rule split).

Top patterns:
- "listitem" role required on empty containers (ul, ol, menu)
- Cannot overwrite role on input elements
- ARIA state disallowed on button role

**Comment by YusukeHirao** [comment]
## Additional nu-validator analysis (post #3645 split)

After splitting wai-aria into sub-rules (#3680), the nu-validator benchmark reveals that #3588 covers two distinct sub-rules:

### `wai-aria-permitted-roles` — 69 violations
`input[type=reset/submit/image]` に対して `getPermittedRoles()` が type 別の permitted roles を正しく返さない。これは Issue 本文で既に記述済み。

### `wai-aria-disallowed-props` — 110+ violations (`aria-expanded` alone)
`aria-expanded` が 37 以上のロール（alert, alertdialog, article, banner, complementary, contentinfo, definition, document 等）で disallowed と判定される。nu-validator は許可。

原因: markuplint の ARIA spec データで、これらのロールの `ownedProperties` に `aria-expanded` が含まれていない。ARIA 1.2 では `aria-expanded` は多くのロールに inherited されるが、spec-generator がこれを反映していない可能性。

`aria-checked` on button role (22 violations) も同様のパターン。

**Comment by YusukeHirao** [comment]
## nu-validator 分析の追加情報（#3680 分割後）

wai-aria 分割後の nu-validator 互換性レポートで、#3588 は **2 つのサブルールに連鎖的に影響** していることが判明。

### `wai-aria-permitted-roles` — 72 violations
`input[type=reset/submit]` の permitted roles が `false`（暗黙ロールのみ許可）のため、ロール上書きが全て不許可。nu-validator は許可。Issue 本文で既に記述済み。

### `wai-aria-disallowed-props` — ~41 violations（連鎖）
`permitted-roles` がロール上書きを拒否 → `getComputedRole()` が暗黙ロール `button` を返す → `aria-checked`, `aria-valuenow`, `aria-selected` 等が `button` ロールで disallowed と判定。

**根本原因は permitted-roles のみ。** input の type 別 permitted roles を修正すれば、`getComputedRole()` が正しいロール（checkbox, slider 等）を返し、これらの property は当然許可される。

### 修正による改善見込み
**~113 件の nu-validator 不一致が解消**（permitted-roles 72 + 連鎖 disallowed-props 41）

**Comment by YusukeHirao** [comment]
v5.0 リリースブロッカー（A群）として #3684 に整理。false positive ~113件の最大ボトルネック。
